src/lib/optimizer/strategy/abstract_rule.hpp

```diff
@@ -29,7 +29,7 @@ class AbstractRule {
    * IMPORTANT: Takes a copy of the node ptr because applying this rule to inputs of this node might remove this node
    * from the tree, which might result in this node being deleted if we don't take a copy of the shared_ptr here.
    */
-  bool _apply_to_inputs(std::shared_ptr<AbstractLQPNode> node);
+  bool _apply_to_inputs(const std::shared_ptr<AbstractLQPNode>& node);
```

Comment:
This was passed by value intentionally, see comment above.

Reply:
Oups, thanks.